### ISSUE TITLE
input seq has diff len for each patient

### ISSUE THREAD

**Issue by bugface** [uf-hobi-informatics-lab/seqEHR#16]
if each patient has a different len of observation, it will raise an error when making tensor in data_utils.py in the mix model.  We need to fix this.

Two proposals:

1. masking - need to change LSTM and TLSTM source code to include mask tensors and masking the gradients
2. padding seq for making tensor then add collate_fn to remove all paddings

**Comment by bugface** [comment]
proposed method for proposal 2:

```python
d = load_pkl("static_seq_lable_test_set_mul.pkl")

nd = []
xnd = []

for i, each in enumerate(d[:5]):
    nd.append((each[0], each[1][:(i+1)*2, :], each[2]))
    xnd.append(each[1][:(i+1)*2, :])

nonseq, seq, label = [], [], []

for each in nd:
    nonseq.append(each[0])
    seq.append(each[1])
    label.append(each[2])
    
ds = TensorDataset(
                tensor(nonseq, dtype=float32),
                torch.nn.utils.rnn.pad_sequence(
                                 [tensor(s,  dtype=float32) for s in seq], 
                                 batch_first=True, 
                                 padding_value=-100),
                tensor(label, dtype=long)
            )

def collate_fn(batch):
    tts = []

    # Gather in lists, and encode labels as indices
    for each in batch:
        n, s, l = each
        t = []
#         mask = ~each[0].eq(-100)
        for e in s:
            if e[0] == -100:
                continue
            t.append(e.numpy())
        tts.append((n, tensor(t), l))
#         ntt = torch.masked_select(each[0], mask)
 
    return tts

dl =  DataLoader(ds, sampler=SequentialSampler(ds), batch_size=1, pin_memory=True, collate_fn=collate_fn)

```

This may require batch fix at 1. We need to test batch size.
